alf/drivers/off_policy_driver.py

```diff
@@ -292,13 +281,15 @@ def _update(self, experience, weight):
                 lambda state: state[0, ...], experience.state)
         else:
             first_train_state = initial_train_state
-        num_steps = experience.step_type.shape[0]
+        num_steps = tf.shape(experience.step_type)[0]
 
         def create_ta(s):
+            # TensorArray cannot use Tensor (batch_size) as element_shape
+            ta_batch_size = experience.step_type.shape[1]
```

Comment:
Can we directly do int(batch_size) here?

Reply:
I didn’t try. But I guess not.

Reply:
> Can we directly do int(batch_size) here?

can not convert a tensor to an integer in graph mode